server/queues/processors/WebsocketsProcessor.ts

```diff
@@ -466,153 +480,172 @@ export default class WebsocketsProcessor {
 
       case "groups.add_user": {
         // do an add user for every collection that the group is a part of
-        const groupMemberships = await GroupMembership.findAll({
+        const groupUser = await GroupUser.findOne({
           where: {
             groupId: event.modelId,
+            userId: event.userId,
           },
         });
-
-        for (const groupMembership of groupMemberships) {
-          // the user being added isn't yet in the websocket channel for the collection
-          // so they need to be notified separately
-          socketio.to(`user-${event.userId}`).emit("collections.add_user", {
-            event: event.name,
-            userId: event.userId,
-            collectionId: groupMembership.collectionId,
-          });
-          // let everyone with access to the collection know a user was added
-          socketio
-            .to(`collection-${groupMembership.collectionId}`)
-            .emit("collections.add_user", {
-              event: event.name,
-              userId: event.userId,
-              collectionId: groupMembership.collectionId,
-            });
-          // tell any user clients to connect to the websocket channel for the collection
-          socketio.to(`user-${event.userId}`).emit("join", {
-            event: event.name,
-            collectionId: groupMembership.collectionId,
-          });
+        if (!groupUser) {
+          return;
         }
 
-        return;
-      }
+        socketio
+          .to(`team-${event.teamId}`)
+          .emit("groups.add_user", presentGroupUser(groupUser));
 
-      case "groups.remove_user": {
-        const groupMemberships = await GroupMembership.findAll({
-          where: {
-            groupId: event.modelId,
-          },
+        socketio.to(`user-${event.userId}`).emit("join", {
+          event: event.name,
+          groupId: event.modelId,
         });
 
-        for (const groupMembership of groupMemberships) {
-          // if the user has any memberships remaining on the collection
-          // we need to emit add instead of remove
-          const collection = groupMembership.collectionId
-            ? await Collection.scope({
-                method: ["withMembership", event.userId],
-              }).findByPk(groupMembership.collectionId)
-            : null;
-
-          if (!collection) {
-            continue;
-          }
+        await GroupMembership.findAllInBatches<GroupMembership>(
+          {
+            where: {
+              groupId: event.modelId,
+            },
+            limit: 100,
```

Comment:
Unrelated but I think `batchSize` here makes more sense than `limit`. Could be taken up separately if we intend to.

Reply:
Perhaps, or `batchLimit`